### ISSUE TITLE
[Error] App exported to Steam OS 3.3.1 (recoll) crashes on launch. Runs properly inside the container (ubuntu-latest)

### ISSUE THREAD

**Issue by ASmellie** [89luca89/distrobox#463]
**Describe the bug**
Application "recoll" crashes on  launch when exported to the host. Runs correctly in the guest
Host:
    Operating System: SteamOS 3.3.1
    KDE Plasma Version: 5.23.5
    KDE Frameworks Version: 5.90.0
    Qt Version: 5.15.2
    Kernel Version: 5.13.0-valve21.1-1-neptune-02211-gc54cda5a36f3 (64-bit)
    Graphics Platform: X11
    Processors: 8 × AMD Custom APU 0405
    Memory: 14.5 GiB of RAM
    Graphics Processor: AMD Custom GPU 0405
Guest container:
   Ubuntu-latest

**To Reproduce**
1. Create a new container
2. Install recoll with
        'apt-get install recoll'
4. Export the app with:
       distrobox-export --app recoll

Launch recoll on the host

App icon shows briefly on the taskbar then disappears

**Expected behavior**
The recoll app should run and show its UI

**Logs**
Find attached the files placed on the host to run recoll in the container
[recoll.zip](https://github.com/89luca89/distrobox/files/9735270/recoll.zip)

**Desktop (please complete the following information):**
Podman 4.2.1
Distrobox 1.4.1

**Comment by BottomNotch** [comment]
I'm having this same issueon SteamOS but with an Arch container.  I enabled "open terminal" on the .desktop file and saw the following error:
`Missing dependency: we need a container manager.
Please install one of podman or docker.
You can follow the documentation on:
        man distrobox-compatibility
or:
        https://github.com/89luca89/distrobox/blob/main/docs/compatibility.md

An error occurred`

podman is installed and on my `$PATH`, and when I copy and paste the command from the .desktop file into konsole, it doesn't have a problem.

**Comment by BottomNotch** [comment]
So here's the workaround to get exported apps working, using Firefox as an example.  Just edit the exec line in the .desktop files like so.
```
Exec=bash -c "xhost +si:localuser:$USER;export PATH=$HOME/.local/bin:$HOME/.local/podman/bin:$PATH;/home/deck/.local/bin/distrobox-enter  -n archlinux-latest -- '  /usr/lib/firefox/firefox  %u'"
```
Basically, the desktop files don't source your .bashrc, so you have to do everything you do for distrobox in .bashrc in the exec line of .desktop files.

**Comment by boxxcar** [comment]
The work around I found was to add the $PATH to the KDE environment variables. So under `$HOME/.config/plasma-workspace/env/` make a script `path.sh`. Inside is the line `export PATH=${PATH}:/home/deck/.local/podman/bin:/home/deck/.local/bin`. I then added `[ ! $(echo $PATH | grep -E '/home/deck/.local') ] && source $HOME/.config/plasma-workspace/env/path.sh` to my .bashrc to remove double entries inside Konsole.

**Comment by BottomNotch** [comment]
> The work around I found was to add the $PATH to the KDE environment variables. So under `$HOME/.config/plasma-workspace/env/` make a script `path.sh`. Inside is the line `export PATH=${PATH}:/home/deck/.local/podman/bin:/home/deck/.local/bin`. I then added `[ ! $(echo $PATH | grep -E '/home/deck/.local') ] && source $HOME/.config/plasma-workspace/env/path.sh` to my .bashrc to remove double entries inside Konsole. E: Source: https://userbase.kde.org/Session_Environment_Variables

That's much better than the way I was doing it, thank you

**Comment by 89luca89** [comment]
@boxxcar thanks for debugging this!

We'll need to find a better place to put stuff than `~/.bashrc` then :smile: 

Can anyone test if putting the PATH stuff in `~/.profile` or `~/.xinitrc` instead?

https://github.com/89luca89/distrobox/blob/main/docs/posts/install_rootless.md

**Comment by BottomNotch** [comment]
> @boxxcar thanks for debugging this!
> 
> We'll need to find a better place to put stuff than `~/.bashrc` then 😄
> 
> Can anyone test if putting the PATH stuff in `~/.profile` or `~/.xinitrc` instead?
> 
> https://github.com/89luca89/distrobox/blob/main/docs/posts/install_rootless.md

.xinitrc didn't work for me, I can try .profile later.  KDE environment variables work great, but I don't think that'll work from gamemode.

**Comment by 89luca89** [comment]
Yea @BottomNotch 
They won't and additionally they are not generic enough for all devices/environments/distros to suggest

**Comment by BottomNotch** [comment]
> Yea @BottomNotch They won't and additionally they are not generic enough for all devices/environments/distros to suggest

My bad, I forgot there's uses for distrobox with rootless podman other than the steam deck

**Comment by BottomNotch** [comment]
> @boxxcar thanks for debugging this!
> 
> We'll need to find a better place to put stuff than `~/.bashrc` then 😄
> 
> Can anyone test if putting the PATH stuff in `~/.profile` or `~/.xinitrc` instead?
> 
> https://github.com/89luca89/distrobox/blob/main/docs/posts/install_rootless.md

.profile doesn't work for me either

**Comment by 89luca89** [comment]
For ~/.profile to work, you'll need to logout and login
Probably easier to reboot

**Comment by BottomNotch** [comment]
> For ~/.profile to work, you'll need to logout and login Probably easier to reboot

I did log out and log back in.  What does work is .xsession, but not in gamemode, I'm trying to figure out if there's something equivalent to .xsession for steamOS gamemode

**Comment by 89luca89** [comment]
Hi @BottomNotch 
Did you find anything about?

**Comment by 89luca89** [comment]
@BottomNotch I'm leaving this open if someone else finds this
Has there been any update on this?

**Comment by NightHammer1000** [comment]
For Wayland alternatives the GNOME Wiki recommends the following:
I quote:

```
Environment variables
The simplest way to set environment variables for a single user under systemd (version 233 or later), is to create a .conf file in ~/.config/environment.d, containing lines of the form NAME=VAL. See the [environment.d documentation](https://www.freedesktop.org/software/systemd/man/environment.d.html) for more details.
```

https://wiki.gnome.org/Initiatives/Wayland/SessionStart

**Comment by NightHammer1000** [comment]
Nevermind.
Found the Solution.
Use /etc/profile

The /etc folder is persistent in SteamOS.

Add `export PATH=$HOME/.local/bin:$HOME/.local/podman/bin:$PATH`
at the end.

Now Exported Apps work Out the Box in Desktop and Gamemode

**Comment by 89luca89** [comment]
Yea I think at this point I'll just update the guide to use `/etc/profile`
It's not something I can do rootlessy and persistently (in /home) from a script that is universal enough for anyone (also non-steamos users)

**Comment by dtantono** [comment]
@89luca89 yeah I tried that workaround. But, it does not worked. what do you think?
[2023-02-25 23-52-12.zip](https://github.com/89luca89/distrobox/files/10831453/2023-02-25.23-52-12.zip)

**Comment by Doc-4** [comment]
From issue #606
> Adding this line to etc/profile didn't result in any change. (I did reboot between adding the line and testing the export) ![image](https://user-images.githubusercontent.com/60247583/221373103-c02fab07-e6a2-4667-a110-fc35a789e692.png)
> 
> I also tried adding it to ~/.bashrc, where the other distrobox run at boot commands on my machine live (it's not there now, I added the picture so you could see what else is being run at boot): ![image](https://user-images.githubusercontent.com/60247583/221373255-8fa481a4-f534-4129-b635-3a860ea18327.png)
> 
> Still no change.

**Comment by 89luca89** [comment]
Hi, I was thinking we could probably find a way to have it working

What about adding:

```sh
export PATH=$HOME/.local/bin:$PATH
export PATH=$HOME/.local/podman/bin:$PATH
xhost +si:localuser:$USER
```

to your `~/.distroboxrc`?

From some user's testing in the Telegram group, it seems to be successful

**Comment by dtantono** [comment]
much better. But, electron based app like Signal -desktop does not worked.

**Comment by 89luca89** [comment]
> much better. But, electron based app like Signal -desktop does not worked (even with flags).

This could be related to missing dependencies or additional flags to pass
I've had some of those problems with electron apps in the past, sadly not fixable by distrobox itself

**Comment by dtantono** [comment]
Yeah, But, sadly electron based apps are very famous like Signal messenger and many people still use it right now. So, I hope this can be fixed and BTW,  I don't think there are missing dependencies because I can run it properly via Konsole

**Comment by Doc-4** [comment]
> Hi, I was thinking we could probably find a way to have it working
> 
> What about adding:
> 
> ```shell
> export PATH=$HOME/.local/bin:$PATH
> export PATH=$HOME/.local/podman/bin:$PATH
> xhost +si:localuser:$USER
> ```
> 
> to your `~/.distroboxrc`?
> 
> From some user's testing in the Telegram group, it seems to be successful
> 
> This should also fix apps launched in Gamemode

I created a new file called .distroboxrc in home and added those lines to it, same exact output. I'm convinced this issue is separate, but you closed my original one. 
Please look into this, it seems like the program has trouble reading files that exist in the directory but only in certain cases, as you can read the log and see that it only errors near thfile:///home/deck/Documents/log.txte end. It had a similar issue installing a dependency it physically shipped with during install, I corrected this by installing both dependencies myself. I've attached the terminal output from after testing this for your convenience.
[log.txt](https://github.com/89luca89/distrobox/files/11072174/log.txt)

**Comment by 89luca89** [comment]
This should be fixed as of 1.5.0.2
